internal/gatewayapi/testdata/merge-valid-multiple-gateways.in.yaml

```diff
@@ -0,0 +1,37 @@
+envoyproxy:
+  apiVersion: gateway.envoyproxy.io/v1alpha1
+  kind: EnvoyProxy
+  metadata:
+    namespace: envoy-gateway-system
+    name: test
+  spec:
+    mergeGateways: true
+gateways:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    metadata:
+      name: gateway-1
+      namespace: envoy-gateway
+    spec:
+      gatewayClassName: envoy-gateway-class
+      listeners:
+        - name: http
```

Comment:
can we add some more gateways or listeners in here to test where only one of the three tuple values (port, proto, hostname) is different, yet those are valid ?